### DISCUSSION TITLE
Unlimited images

### DISCUSSION THREAD

**Discussion by jordanbr20** [lllyasviel/Fooocus#1511] [Q&A] [answered]
Is there a way to extend the 32 limit of images to sometimes a lot larger?

**Comment by mashb1t** [comment]
Feel free to adjust the maximum in https://github.com/lllyasviel/Fooocus/blob/main/webui.py#L229C84-L229C84 and increase the limit to whatever value you like. Code changes will only take effect after restarting the application.

**Reply by adam-ah** [reply]
@mashb1t If there is no obvious reason to limit to 32 (I haven't found one), could you please make a PR to increase to something no one will reach, like 1600? 32 as a default max is rather low for me too, I'd prefer to leave the computer overnight generating images.

**Reply by mashb1t** [reply]
This was already part of https://github.com/lllyasviel/Fooocus/pull/1248, which has been closed.
What you always can do is to right-click the "Generate" button to then click "generate forever".

**Reply by mashb1t** [reply, marked as answer]
I've extracted this part of the closed PR and created a new one:  https://github.com/lllyasviel/Fooocus/pull/1616

**Reply by adam-ah** [reply]
Would it be possible to be able to enter manually and not to "validate" the number, similar to what koboldcpp lite UI does? It has a slider, but manually entering a number can move the value outside of the slider. Ultimately, it's my rendering, my choice, the software should not dictate how many I need

**Reply by mashb1t** [reply]
There already is an input field in every Gradio slider component, which you can use by clicking on the number next to it.
The maximum is useful for server administrators so users don't occupy the queue for an indefinite amount of time, also when using Fooocus in a hosted environment with many users in parallel.
Feel free to adjust the config on your local machine to your liking though, could be 10k+ or whatever you desire. The goal of this PR is to make this possible and not to lock people into a maximum of 32 (hardcoded currently), so we totally agree on the key point 👍

**Reply by adam-ah** [reply]
I appreciate you sharing your thoughts and working towards eliminating the use of hardcoded magic numbers. However, I believe that an arbitrary number like 32 might not make sense to everyone.

For instance, on the sdxl-turbo model, I can generate 32 images in just a few seconds. But, for someone using a shared environment with a 3060 and the full sdxl model with quality settings, it might take the proverbial "eternity" to render the same number of images. Here, too, 32 seemed like an arbitrary number.

While I agree that limiting the number of images for a shared setting might be helpful (although only a very small fraction of users might need this), I think a separate runner could be a better solution than limiting it to 32 for everyone with all models, configurations, and environments.

For shared environments, perhaps a time limit might be more effective, while a no hard-limit might work better for most users?

**Reply by mashb1t** [reply]
Maybe we misunderstood another: The PR removes the currently implemented limit and only uses 32 as a default value ready to be modified using the config. This isn't possible before the PR is merged without having to do code adjustments.
My aim is to not limit anybody to an arbitrary number of images but make it customizable.

EDIT: setting a limit by default makes sense as Gradio gets out of sync as each additional image in the temporary rendered results takes more time to display and Gradio beginns to decline in display time, leading to UI updates to lag behind renders quite a bit. Please also find the solution to this in the PR description.
